Various updates.

davidlehn opened the pull request:
- Some of these commits can be easily merged, just doing some cleanup, and using our other karma coverage support.
- Testing issues:
  - The karma test and coverage tests are failing for me.
  - node 16 and npm 8.14.0 are installing things in the wrong places sometimes.  often putting or reorganizing deps deep in the tree, and other tools then fail to find things.  for instance, css-loader multiple times deep in the tree and it can't be found.  or karma-webpack under @bedrock/karma so the plugin isn't found.
  - The main lib is failing to find the peer deps in the test dir.

I assume I have something setup wrong, but I'm not sure what.

Should libs like this have .npmrc with legacy-peer-deps option?

davidlehn committed:
Use test karma coverage commands.

davidlehn committed:
Update eslint support.

- Update eslint packages.
- Lint module.

davidlehn committed:
Add changelog.

davidlehn committed:
Update dependencies.

davidlehn committed:
Update files field.

davidlehn committed:
Remove unused bedrock flag.

davidlehn committed:
Move dependencies.

davidlehn committed:
Enable testing.

davidlehn commented:
As an example, CI system failing to find karma-webpack because it was probably installed not at the top level: https://github.com/digitalbazaar/bedrock-vue-vc/runs/7363191289?check_suite_focus=true

dlongley commented:
@davidlehn,

> Should libs like this have .npmrc with legacy-peer-deps option?

I didn't see it on `bedrock-vue-wallet` but maybe it should also be in all the `bedrock-vue*` libs. I'm guessing we should add it for symmetry at the moment and then back it all out later if we're able.

davidlehn commented:
@dlongley Are you approving so as to get this into initial where it will still be broken?  I'm not sure how to fix the npm oddness here.

tolson4 commented on components/credentialCommon.js:
```diff
@@ -3,6 +3,10 @@
+/**
+ * @typedef Ref
+ */
+
```
Should be able to drop this as it was addressed in #1 as part of this [commit](https://github.com/digitalbazaar/bedrock-vue-vc/pull/1/commits/4a1122145b9b4b1c9023ab51062930e14c2fe365#diff-8942e9f15d3b88aeed7de27c6d94aa1345f4217fb879ab73053732f2a4b78e76)

tolson4 commented:
> Testing issues:
> * The karma test and coverage tests are failing for me.
> * node 16 and npm 8.14.0 are installing things in the wrong places sometimes.  often putting or reorganizing deps deep in the tree, and other tools then fail to find things.  for instance, css-loader multiple times deep in the tree and it can't be found.  or karma-webpack under @bedrock/karma so the plugin isn't found.
> * The main lib is failing to find the peer deps in the test dir.

@davidlehn For what its worth, I had a lot of the same issues locally. I tried for a bit to work around them in order to update the unit tests (I'm not sure if they'll pass or not) but couldn't get past these issues.

davidlehn committed:
Add npmrc with legacy-peer-deps option.

aljones15 committed:
Add aliases for vue & @bedrock/web.

aljones15 committed:
Fix renderCredential.

tolson4 committed:
Update and fix tests.

tolson4 committed:
Lint.

davidlehn committed:
Lint all source files.

davidlehn committed:
Fix lint issues.

davidlehn committed:
Improve aliases.

- Fix alias matching pattern.
- Remove direct package file accesses.
  - Avoids assuming internal package structure.
  - Let's ESM versions be used automatically.
- Update vue and quasar imports.

davidlehn committed:
Disable coverage.

- vue support not currently working.

davidlehn commented:
- I don't fully understand why that karma alias config is needed, but that's an issue for another day.
- Skipping coverage for now since vue support does not yet work. Unfortunate, since it's hard to tell what the coverage may be.
  - https://github.com/digitalbazaar/bedrock-karma/issues/24
- There seem to be test files that are not used like `test/coverage/index.js`

Is this ok to merge to `initial`?

tolson4 commented on components/CredentialBase.vue:
```diff
@@ -75,10 +75,10 @@
 /*!
  * Copyright (c) 2018-2022 Digital Bazaar, Inc. All rights reserved.
  */
-import {computed, defineProps, reactive, toRef, unref} from 'vue';
```
`defineProps` was explicitly left in even though it is not required. There is some more discussion on this here: https://github.com/digitalbazaar/chapi-playground/pull/1#discussion_r894022268

I am unsure if a standard has been chosen for DB on this going forward, but have been including `defineEmits` and `defineProps` as needed to provide clarity on what they are and where they are coming from.

davidlehn replied:
Ok, I dropped that commit.  I didn't see that discussion before and was just following the note on the console.  It is very likely people omit this again in the future, here or elsewhere.  I didn't see any rules in https://eslint.vuejs.org/ that would enforce the import and a soft policy won't work so well for this situation.

cc: @dlongley

tolson4 replied:
We can continue the discussion here https://github.com/digitalbazaar/chapi-playground/pull/1#discussion_r894022268

Once there is some consensus on the standard going forward I will go ahead and come back and update things if needed. In the meantime we can go ahead and merge this into #1

davidlehn committed:
Simplify import path.
